src/lightning/pytorch/utilities/exceptions.py

```diff
@@ -33,10 +33,13 @@ class _TunerExitException(Exception):
 
 
 def _augment_message(exception: BaseException, pattern: str, new_message: str) -> None:
-    if _PYTHON_GREATER_EQUAL_3_11_0 and any(re.match(pattern, message, re.DOTALL) for message in exception.args):
+    # Remove this when Python 3.11 becomes the minimum supported version
+    if not _PYTHON_GREATER_EQUAL_3_11_0:
+        args = []
+        for arg in exception.args:
+            if isinstance(arg, str) and re.match(pattern, arg, re.DOTALL):
+                arg = new_message
+            args.append(arg)
+        exception.args = tuple(args)
```

Comment:
Could we write this as a list-comprehension as done previously? If I see correctly, the addition you made was checking for a string instance.